### PR TITLE
add web-view

### PR DESCRIPTION
**Pull request by dingsheng-ong**

### CODE REVIEW

**Comment by dingsheng-ong** [comment]
It seems like the smoke test failed because the `web-view` has not yet been merged. @fjebaker

**Comment by dingsheng-ong** [comment]
> It seems like the smoke test failed because the `web-view` has not yet been merged. @fjebaker

I'm self-merging the `web-view` repo since I was the first to commit to it. The test passed as soon as I merged the pull request in the `web-view` repo.

File: database-setup/initfile.sql [hunk 1]
```diff
@@ -2,6 +2,20 @@ CREATE DATABASE IF NOT EXISTS llanwrytd;
 
 USE llanwrytd;
 
+CREATE TABLE userData (
+    userID VARCHAR(128) NOT NULL,
+    userEmail VARCHAR(128),
+    userPassword VARCHAR(256),
+    userRole ENUM('admin', 'non-admin', 'pending'),
+    PRIMARY KEY ( userID )
```
Comment (fjebaker) [inline database-setup/initfile.sql:10]: Could we use an auto-incrementing `INT` for the primary key? Has a few benefits, the least of which being easier range selection, and a more distinguishable ID (cf: USER, US3R).

File: database-setup/initfile.sql [hunk 2]
```diff
@@ -2,6 +2,20 @@ CREATE DATABASE IF NOT EXISTS llanwrytd;
...
+);
+
+INSERT INTO userData SET
+    userID = 'admin',
+    userEmail = 'admin@aimlac.com',
+    userPassword = '71f3fcb0ee1407a3c49a9b2f207f68c9f21cd6dec7a516499883b4c740d01295',
```
Comment (fjebaker) [inline database-setup/initfile.sql:16]: What does this hash decode to? Could add as a comment?

File: docker-compose.yml [hunk 3]
```diff
@@ -54,6 +55,32 @@ services:
...
+  nginx:
+    image: nginx:1.21.6-alpine
+    networks:
+      - backend
+    depends_on:
+      - web-view
+    ports:
+      - 80:80
+    volumes:
+      - "${PWD}/web-view/nginx:/etc/nginx/conf.d"
```
Comment (fjebaker) [inline docker-compose.yml:82]: See my messages on discord about this.

File: database-setup/initfile.sql (hunk 2 above)
Comment (dingsheng-ong) [reply database-setup/initfile.sql:16]: that's the SHA256 hash for the password (`aimlac`), will add this to the file

File: docker-compose.yml (hunk 3 above)
Comment (dingsheng-ong) [reply docker-compose.yml:82]: 👍 , will move the nginx config into this repo

File: database-setup/initfile.sql (hunk 1 above)
Comment (dingsheng-ong) [reply database-setup/initfile.sql:10]: Sure, I can add a column named `userIndex` and make it the primary key.

Comment (fjebaker) [reply database-setup/initfile.sql:10]: Sounds good, feel free to merge afterwards 👍

Comment (dingsheng-ong) [reply database-setup/initfile.sql]: issue resolved in the commit [4b12c77](https://github.com/aimlac-7564616D/composer/pull/9/commits/4b12c774926b6147fc12359b1b68d2aadd97c1d2)

**Review by fjebaker — APPROVED** [review]

**Pull request merged**